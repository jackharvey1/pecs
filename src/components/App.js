import React, { Component } from 'react';
import styled from 'styled-components';
import images from '../assets';
import PictureBoard from './composites/PictureBoard';

const SuperContainer = styled.div`
    width: 100vw;
    height: 100vh;

    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
`;

export default class App extends Component {
    constructor () {
        super();

        const pictureBank = Object.keys(images)
            .map(imageName => ({
                src: images[imageName],
                alt: imageName,
            }));

        this.state = {
            dragging: false,
            pictureBank,
            storyboard: [],
        };
    }

    render () {
        return (
            <SuperContainer>
                <PictureBoard
                    centered
                    id="storyboard"
                    data={this.state.storyboard}
                    onDragStart={this.onDragStart.bind(this)}
                    onDrop={this.onDrop.bind(this)}
                    onDragOver={this.onDragOver.bind(this)}
                    dragging={this.state.dragging}
                />
                <PictureBoard
                    id="pictureBank"
                    data={this.state.pictureBank}
                    onDragStart={this.onDragStart.bind(this)}
                    onDrop={this.onDrop.bind(this)}
                    onDragOver={this.onDragOver.bind(this)}
                />
            </SuperContainer>
        );
    }

    onDragStart (event, src) {
        const [eventSource, sourceIndex] = event.target.id.split('-');

        const { alt } = this.state.pictureBank.find((image) => image.src === src);
        const sourceInfo = { eventSource, sourceIndex, src, alt };
        event.dataTransfer.setData('sourceInfo', JSON.stringify(sourceInfo));
    }

    onDrop (event) {
        event.preventDefault();

        const sourceInfo = event.dataTransfer.getData('sourceInfo');
        const { eventSource, sourceIndex, src, alt } = JSON.parse(sourceInfo);

        const [eventDestination, destinationIndex] = event.target.id.split('-');

        const storyboardClone = this.state.storyboard.slice();

        if (eventDestination === 'pictureBank' || eventDestination === 'storyboard') {
            if (eventSource === 'storyboard') {
                storyboardClone.splice(sourceIndex, 1);
            }

            if (eventDestination === 'storyboard') {
                storyboardClone.splice(destinationIndex, 0, { src, alt });
            }
        }

        this.setState({
            storyboard: storyboardClone,
            dragging: false,
        });
    }

    onDragOver (event) {
        event.preventDefault();

        if (!this.state.dragging) {
            this.setState({
                dragging: true,
            });
        }
    }
}
