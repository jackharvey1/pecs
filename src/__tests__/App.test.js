import React from 'react';
import renderer from 'react-test-renderer';
import App from '../components/App';

it('renders renders as intended', () => {
    const wrapper = renderer.create(<App />).toJSON();
    expect(wrapper).toMatchSnapshot();
});
