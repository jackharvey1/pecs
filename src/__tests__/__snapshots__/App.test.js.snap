// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders renders as intended 1`] = `
.c4 {
  color: #333333;
  font-size: 16px;
  font-family: Helvetica,Arial,sans-serif;
  margin-top: 12px;
  text-align: center;
}

.c3 {
  box-shadow: 0 2px 8px 0 rgba(0,0,0,.25);
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 20px;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c0 {
  width: 100vw;
  height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <img
        alt="breakfast"
        className="c3"
        src="breakfast.png"
      />
      <p
        className="c4"
      />
    </div>
    <div
      className="c2"
    >
      <img
        alt="getReady"
        className="c3"
        src="get-ready.png"
      />
      <p
        className="c4"
      />
    </div>
    <div
      className="c2"
    >
      <img
        alt="shoes"
        className="c3"
        src="shoes.png"
      />
      <p
        className="c4"
      />
    </div>
    <div
      className="c2"
    >
      <img
        alt="timedVideoGame"
        className="c3"
        src="timed-video-game.png"
      />
      <p
        className="c4"
      />
    </div>
    <div
      className="c2"
    >
      <img
        alt="toysTidiedUp"
        className="c3"
        src="toys-tidied-up.png"
      />
      <p
        className="c4"
      />
    </div>
    <div
      className="c2"
    >
      <img
        alt="washYourHands"
        className="c3"
        src="wash-your-hands.png"
      />
      <p
        className="c4"
      />
    </div>
  </div>
</div>
`;
